### PR TITLE
Replace Applicable Zod validations with those from Regtech-Regex

### PR DESCRIPTION
**Pull request by tanner-ricks**
Added dependency to regtech-regex and updated the zod stuff to use the regexs held within.

## Changes

- Updated dependencies
- Added dependency to regtech-regex
- Updated zod validations with the regexes from regtech-regex

## How to test this PR

1. Validate that the pipeline is green
2. Pull the branch
3. Run yarn install
4. Launch the app
5. Create a new filing and validate that you are able to reach the submission page

### CODE REVIEW

File: src/types/formTypes.ts [hunk 1]
```diff
@@ -109,15 +113,15 @@ export const institutionDetailsApiTypeSchema = z.object({
       z.number({
         invalid_type_error: IdZodSchemaErrors.parent_rssd_idNumber,
       }),
-      z
-        .string()
-        .regex(/^\d+$|^$/, { message: IdZodSchemaErrors.parent_rssd_idRegex }),
```
Comment (billhimmelsbach) [inline src/types/formTypes.ts:114]: Looks like this regex also allows empty string, so replacing it with the regtech-regex one causes a bug where I can't submit an Update your financial institution form with a blank RSSD ID ("Hey CFPB, remove this RSSD ID"). 

Screenshot 2024-12-20 at 1 58 24 PM

Compare this with trying to submit with a blank TIN. Maybe we can use the regtech-regex regex but make this field nullable/optional?

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]
Looking great! One small thing to fix, but then this is ready to go.

File: src/types/formTypes.ts (hunk 1 above)
Comment (tanner-ricks) [reply src/types/formTypes.ts:114]: Should be taken care of.

**Review by billhimmelsbach — APPROVED** [review]
That works! Thanks.

**Pull request merged**